src/enclave-agent/src/config.rs

```diff
@@ -0,0 +1,136 @@
+            File::create(Path::new(&path))?.write_all(DEFAULT_KEYPROVIDER_CONFIG.as_bytes())?;
+            kc.is_exist = true;
+        }
+        Ok(kc)
+    }
+
+    pub fn export_to_env(&self) {
+        std::env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", self.config_path.clone());
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_load_decrypt_config() {
+        struct ConfigCase {
+            path: String,
+            load_fail: bool,
+            result_key_provider: Option<String>,
+            result_security_validate: Option<bool>,
+        }
+        let decrypt_opt = "provider:attestation-agent:eaa_kbc::127.0.0.1:1234";
+        let config_dir = "test_data/decrypt_config";
+        let config_cases: Vec<ConfigCase> = vec![
+            ConfigCase {
+                path: format!("{}/{}", config_dir, "decrypt_config_not_existing.conf"),
+                load_fail: true,
+                result_key_provider: Some(decrypt_opt.to_string()),
+                result_security_validate: Some(true),
+            },
+            ConfigCase {
+                path: format!("{}/{}", config_dir, "decrypt_config_format_invalid.conf"),
+                load_fail: true,
+                result_key_provider: Some(decrypt_opt.to_string()),
+                result_security_validate: Some(true),
+            },
+            ConfigCase {
+                path: format!("{}/{}", config_dir, "decrypt_config_normal.conf"),
+                load_fail: false,
+                result_key_provider: Some(decrypt_opt.to_string()),
+                result_security_validate: Some(true),
+            },
+            ConfigCase {
+                path: format!("{}/{}", config_dir, "decrypt_config_disable_validate.conf"),
+                load_fail: false,
+                result_key_provider: Some(decrypt_opt.to_string()),
+                result_security_validate: Some(false),
+            },
+            ConfigCase {
+                path: format!("{}/{}", config_dir, "decrypt_config_default_validate.conf"),
+                load_fail: false,
+                result_key_provider: Some(decrypt_opt.to_string()),
+                result_security_validate: Some(true),
+            },
+            ConfigCase {
+                path: format!("{}/{}", config_dir, "decrypt_config_empty.conf"),
+                load_fail: false,
+                result_key_provider: None,
+                result_security_validate: Some(true),
+            },
+        ];
+
+        for case in config_cases {
+            let result = DecryptConfig::load_from_file(&case.path);
+            if case.load_fail {
+                assert!(result.is_err());
+            } else {
+                let c = result.unwrap();
+                assert_eq!(c.key_provider, case.result_key_provider);
+                assert_eq!(c.security_validate, case.result_security_validate);
+            }
+        }
+    }
+
+    #[test]
+    fn test_keyprovider_config_default() {
```

Comment:
It fails because the test case is not allowed to write to `DEFAULT_KEYPROVIDER_CONFIG_PATH`:

```
---- config::test::test_keyprovider_config_default stdout ----
thread 'config::test::test_keyprovider_config_default' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }', src/config.rs:126:54
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
The test case is passed in our env. The agent should run in the root privilege to write to `/etc`. If we want to place the config file in different place, please propose another issue to track the concern.

Reply:
> The agent should run in the root privilege to write to `/etc`

this is not about running the agent but running the running tests. `cargo test` should pass without root/sudo privileges.

Reply:
If you insist it, please change the [enclave-agent.yaml](https://github.com/confidential-containers/enclave-cc/blob/572a5715bc4941c56e0b4dc0b4dacd3e5a9c4c42/tools/packaging/build/agent-enclave-bundle/enclave-agent.yaml#L25) , copy the ` /run/enclave-agent/decrypt_config.conf` to other directory instead of `/etc`.

Reply:
The keyprovider config will not be writeback, the unit test can be run without root privilege now.